spring-cloud-gateway-core/src/main/java/org/springframework/cloud/gateway/config/GatewayCircuitBreakerAutoConfiguration.java

```diff
@@ -39,15 +39,13 @@
  */
 @Configuration(proxyBeanMethods = false)
 @ConditionalOnProperty(name = "spring.cloud.gateway.enabled", matchIfMissing = true)
-@AutoConfigureAfter({ ReactiveResilience4JAutoConfiguration.class,
-		HystrixCircuitBreakerAutoConfiguration.class })
-@ConditionalOnClass({ DispatcherHandler.class,
-		ReactiveResilience4JAutoConfiguration.class,
-		HystrixCircuitBreakerAutoConfiguration.class })
+@ConditionalOnClass({ DispatcherHandler.class })
 public class GatewayCircuitBreakerAutoConfiguration {
 
 	@Configuration(proxyBeanMethods = false)
-	@ConditionalOnClass({ ReactiveCircuitBreakerFactory.class,
+	@AutoConfigureAfter({ HystrixCircuitBreakerAutoConfiguration.class })
```

Comment:
I'm not positive `@AutoConfigureAfter` works here since this class isn't auto configuration. It may need to be split into two separate auto configuration classes. @snicoll or @philwebb can you confirm my suspicion?

Reply:
You are correct. `@AutoConfigureBefore`/`@AutoConfigureAfter` are used to sort auto-configuration classes before they are processed. Inner classes and `@Imports` are handled by Spring Framework's standard `@Configuration` class processing logic.